sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -1017,6 +1017,9 @@ object TypeCoercion extends TypeCoercionBase {
       case (ArrayType(fromType, fn), ArrayType(toType: DataType, true)) =>
         implicitCast(fromType, toType).map(ArrayType(_, true)).orNull
 
+      case (ArrayType(fromType, fn), AbstractArrayType(toType: StringTypeCollated)) =>
```

Comment:
```suggestion
      case (ArrayType(fromType, fn), AbstractArrayType(toType)) =>
```

Reply:
No expression uses `AbstractArrayType` with non string type now, but it's pretty easy to make it future-proof so let's do it.